Sporadic unresolved test failures

We're encountering sporadic unresolved failures. So far, they've always happened with x86 Clang in this test:

```
1: Unresolved Tests (1):
1:   libc++ :: std/input.output/string.streams/stringbuf/stringbuf.members/view.pass.cpp:2
```

# Test File

https://github.com/llvm/llvm-project/blob/57f42a8765cd3d878be4fb59ad44c85f8a7ca223/libcxx/test/std/input.output/string.streams/stringbuf/stringbuf.members/view.pass.cpp

# Occurrences

Date | PR | Arch | Azure Pipelines Run
--- | --- | --- | ---
2024-01-25 | #4274 | x86 | https://dev.azure.com/vclibs/STL/_build/results?buildId=15987&view=results
2024-01-27 | #4345 | x86 | https://dev.azure.com/vclibs/STL/_build/results?buildId=16004&view=results
2024-02-03 | #4365 | x86 | https://dev.azure.com/vclibs/STL/_build/results?buildId=16061&view=results


# Output

Thanks to #4323, we have some useful output:

```
UnicodeDecodeError('charmap', b'minkernel\\crts\\ucrt\\src\\appcrt\\heap\\debug_heap.cpp(904) : Assertion failed: _CrtIsValidHeapPointer(block)\r\nminkernel\\crts\\ucrt\\src\\appcrt\\heap\\debug_heap.cpp(908) : Assertion failed: is_block_type_valid(header->_block_use)\r\nHEAP CORRUPTION DETECTED: before R\xe8\xa5j\xfe\xff\x83\xc4\x08\x8dE\x08P\x8dM\xf0\xe8F6\xfe\xffPj\x05\xe8\x0e\xfc\xff\xff\x83\xc4\x08\x8dM\x08Q\x8dM\xec\xe8/6\xfe\xffPj\x04\xe8\x87\xfb\xff\xff\x83\xc4\x08\x8bM\xf4d\x89\r block (#-33686019) at 0x00EF2CD4.\nCRT detected that the application wrote to memory before start of heap buffer.\nHEAP CORRUPTION DETECTED: after R\xe8\xa5j\xfe\xff\x83\xc4\x08\x8dE\x08P\x8dM\xf0\xe8F6\xfe\xffPj\x05\xe8\x0e\xfc\xff\xff\x83\xc4\x08\x8dM\x08Q\x8dM\xec\xe8/6\xfe\xffPj\x04\xe8\x87\xfb\xff\xff\x83\xc4\x08\x8bM\xf4d\x89\r block (#-33686019) at 0x00EF2CD4.\nCRT detected that the application wrote to memory after end of heap buffer.\nminkernel\\crts\\ucrt\\src\\appcrt\\heap\\debug_heap.cpp(996) : Assertion failed: __acrt_first_block == header\r\n', 266, 267, 'character maps to <undefined>') at D:\a\_work\1\s\tests\utils\stl\test\format.py:231
```

# Open Questions

* [ ] What's causing this `_CrtIsValidHeapPointer` assertion, which is the underlying spurious failure?
* [ ] Why is this specific to x86 Clang and `stringbuf.members/view.pass.cpp`? Is it a library bug in our `basic_stringbuf`, a test bug, or a compiler bug?
* [ ] Why is the Python machinery encountering a `UnicodeDecodeError`? Is something about the test output, like the heap contents, tripping it up? Do we need to fix the Python code somewhere, or does the UCRT need to fix its output?

I got the same error with the `std/utilities/format/format.functions/escaped_output.ascii.pass.cpp` test.

```
stl-lit.py: D:\git-projects\wg21\STL\tests\utils\stl\test\format.py:250: error: UnicodeDecodeError('gbk', b"Assertion failure: out == expected D:\\git-projects\\wg21\\STL\\llvm-project\\libcxx\\test\\std\\utilities\\format\\format.functions\\escaped_output.ascii.pass.cpp 46\r\n\r\nFormat string   {:?}\r\nExpected output '\x80'\r\nActual output   '\\x{80}'\r\nabort() has been called", 198, 199, 'illegal multibyte sequence') at D:\git-projects\wg21\STL\tests\utils\stl\test\format.py:231
UNRESOLVED: libc++ :: std/utilities/format/format.functions/escaped_output.ascii.pass.cpp:1 (16 of 63)
******************** TEST 'libc++ :: std/utilities/format/format.functions/escaped_output.ascii.pass.cpp:1' FAILED ********************
Exception during script execution:
Traceback (most recent call last):
  File "D:\git-projects/wg21/STL/llvm-project/llvm\utils\lit\lit\worker.py", line 77, in _execute_test_handle_errors
    return _adapt_result(result)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\git-projects/wg21/STL/llvm-project/llvm\utils\lit\lit\worker.py", line 92, in _adapt_result
    assert isinstance(result, tuple)
AssertionError


********************
```

It seems that [`STLTestFormat.execute`](https://github.com/microsoft/STL/blob/3eac329d1f614ecf138d96c22a3b02f87076bc4a/tests/utils/stl/test/format.py#L212) returns `None` when an exception has been raised, but llvm's [`_adapt_result`](https://github.com/llvm/llvm-project/blob/49138d97c0d8a6d1c6935da414a1f3fea839263b/llvm/utils/lit/lit/worker.py#L89) expects it to return something meaningful (either `lit.Test.Result` or `tuple`).

--- a/tests/utils/stl/util.py
+++ b/tests/utils/stl/util.py
@@ -77,8 +77,9 @@ def decodeOutput(bytes):
     try:
         return bytes.decode()
     except UnicodeError:
+        # Use 'backslashreplace' to avoid throwing another exception for unrecognized characters.
         import locale
-        return bytes.decode(locale.getpreferredencoding(do_setlocale=False))
+        return bytes.decode(locale.getpreferredencoding(do_setlocale=False), 'backslashreplace')
 
 
 def executeCommand(command, cwd=None, env=None, input=None, timeout=0):

